src/internal_events/common.rs

```diff
@@ -90,3 +90,36 @@ impl InternalEvent for AwsBytesSent {
         );
     }
 }
+
+#[derive(Debug)]
+pub struct StreamClosedError {
+    pub error: String,
+    pub count: usize,
+}
+
+impl InternalEvent for StreamClosedError {
+    fn emit_logs(&self) {
+        error!(
+            message = "Failed to forward event, downstream is closed.",
```

Comment:
```suggestion
            message = "Failed to forward event(s), downstream is closed.",
```
Maybe a slight bit clearer since this can involve multiple events.